wpilibc/src/main/native/cpp/PS4Controller.cpp

```diff
@@ -0,0 +1,205 @@
+}
+
+bool PS4Controller::GetR1ButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kR1));
+}
+
+bool PS4Controller::GetR1ButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kR1));
+}
+
+bool PS4Controller::GetL2Button() const {
+  return GetRawButton(static_cast<int>(Button::kL2));
+}
+
+bool PS4Controller::GetL2ButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kL2));
+}
+
+bool PS4Controller::GetL2ButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kL2));
+}
+
+bool PS4Controller::GetR2Button() const {
+  return GetRawButton(static_cast<int>(Button::kR2));
+}
+
+bool PS4Controller::GetR2ButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kR2));
+}
+
+bool PS4Controller::GetR2ButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kR2));
+}
+
+bool PS4Controller::GetShareButton() const {
+  return GetRawButton(static_cast<int>(Button::kShare));
+}
+
+bool PS4Controller::GetShareButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kShare));
+}
+
+bool PS4Controller::GetShareButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kShare));
+}
+
+bool PS4Controller::GetOptionsButton() const {
+  return GetRawButton(static_cast<int>(Button::kOptions));
+}
+
+bool PS4Controller::GetOptionsButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kOptions));
+}
+
+bool PS4Controller::GetOptionsButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kOptions));
+}
+
+bool PS4Controller::GetLeftStickButton() const {
+  return GetRawButton(static_cast<int>(Button::kLeftStick));
+}
+
+bool PS4Controller::GetLeftStickButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kLeftStick));
+}
+
+bool PS4Controller::GetLeftStickButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kLeftStick));
+}
+
+bool PS4Controller::GetRightStickButton() const {
+  return GetRawButton(static_cast<int>(Button::kRightStick));
+}
+
+bool PS4Controller::GetRightStickButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kRightStick));
+}
+
+bool PS4Controller::GetRightStickButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kRightStick));
+}
+
+bool PS4Controller::GetPSButton() const {
+  return GetRawButton(static_cast<int>(Button::kPS));
+}
+
+bool PS4Controller::GetPSButtonPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kPS));
+}
+
+bool PS4Controller::GetPSButtonReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kPS));
+}
+
+bool PS4Controller::GetScreen() const {
+  return GetRawButton(static_cast<int>(Button::kScreen));
+}
+
+bool PS4Controller::GetScreenPressed() {
+  return GetRawButtonPressed(static_cast<int>(Button::kScreen));
+}
+
+bool PS4Controller::GetScreenReleased() {
+  return GetRawButtonReleased(static_cast<int>(Button::kScreen));
+}
```

Comment:
Not sure that `Screen` is quite clear enough, had to think for a few moments to figure out what it meant. Maybe `Touchpad`?